scripts/main/albums.js

```diff
@@ -78,68 +82,64 @@ albums.load = function () {
 	}
 };
 
+/**
+ * @param {(Album|TagAlbum|SmartAlbum)} album
+ * @returns {void}
+ */
 albums.parse = function (album) {
 	if (!album.thumb) {
-		album.thumb = {};
-		album.thumb.id = "";
-		album.thumb.thumb = album.has_password ? "img/password.svg" : "img/no_images.svg";
-		album.thumb.type = "";
-		album.thumb.thumb2x = null;
+		album.thumb = {
+			id: "",
+			thumb: album.has_password ? "img/password.svg" : "img/no_images.svg",
+			type: "image/svg+xml",
+			thumb2x: null,
+		};
 	}
 };
 
-// TODO: REFACTOR THIS
-albums._createSmartAlbums = function (data) {
+/**
+ * Normalizes the built-in smart albums.
+ *
+ * @param {SmartAlbums} data
+ * @returns {void}
+ */
+albums.localizeSmartAlbums = function (data) {
 	if (data.unsorted) {
-		data.unsorted = {
-			id: "unsorted",
-			title: lychee.locale["UNSORTED"],
-			created_at: null,
-			is_unsorted: true,
-			thumb: data.unsorted.thumb,
-		};
+		data.unsorted.title = lychee.locale["UNSORTED"];
 	}
 
 	if (data.starred) {
-		data.starred = {
-			id: "starred",
-			title: lychee.locale["STARRED"],
-			created_at: null,
-			is_starred: true,
-			thumb: data.starred.thumb,
-		};
+		data.starred.title = lychee.locale["STARRED"];
 	}
 
 	if (data.public) {
-		data.public = {
-			id: "public",
-			title: lychee.locale["PUBLIC"],
-			created_at: null,
-			is_public: true,
-			requires_link: true,
-			thumb: data.public.thumb,
-		};
+		data.public.title = lychee.locale["PUBLIC"];
+		// TODO: Why do we need to set these two attributes? What component relies upon them, what happens if we don't set them? Is it legacy?
```

Comment:
Legacy and probably can be nuked due to new backend.

Reply:
This has nothing to do with the backend.

But as I found out in the meantime the code in `view.js` (or somewhere related) relies upon the attributes `is_public` and `requires_link` to show the correct buttons in the menu. Of course, this is not how it should be, but unless we fix the view (I believe it was `buildStructure` or something like that), this must remain an open todo.